napari/utils/_pydantic/evented_model.py

```diff
@@ -0,0 +1,67 @@
+from typing import Dict
+
+from pydantic import BaseModel, PrivateAttr
+
+from ..events.dataclass import _type_to_compare, is_equal
+from ..events.event import EmitterGroup
+from .custom_types import JSON_ENCODERS
+
+
+class EventedModel(BaseModel):
+
+    # add private attributes for event emission
+    _events: EmitterGroup = PrivateAttr(default_factory=EmitterGroup)
+    __equality_checks__: Dict = PrivateAttr(default_factory=dict)
+
+    # Definte the config so that assigments are validated
+    # and add custom encoders
+    class Config:
+        arbitrary_types_allowed = True
+        validate_assignment = True
+        underscore_attrs_are_private = True
+        use_enum_values = True
+        validate_all = True
+        json_encoders = JSON_ENCODERS
+
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+
+        # get fields and add to EmitterGroup
+        _fields = list(self.__fields__)
+        e_fields = {fld: None for fld in _fields}
+
+        # add events for each field
+        self.events.add(**e_fields)
```

Comment:
```suggestion
        # add events for each field
        self._events.source = self
        self._events.add(**dict.fromkeys(self.__fields__))
```
note, it's important that `self._events.source = self` come before adding events.  Also, the reason the everything was fine without the `__weakref__` classVar is because no events had sources anymore (which we definitely want, right?)

Reply:
but also ... I _love_ how much cleaner this handles inheritance (compared to `evented_dataclass`) with _events as a `PrivateAttr`